packages/ember/tests/routing/router_service_test/transitionTo_test.js

```diff
@@ -364,6 +364,62 @@ moduleFor(
       });
     }
 
+    async ['@test RouterService#transitionTo should not refresh parent models if the parent query params stay the same'](assert) {
+      let parentModelHookCallCount = 0;
+      this.add(
+        'route:parent',
+        Route.extend({
+          queryParams: {
+            parentParam: {
+              refreshModel: true
+            }
+          },
+          model() {
+            parentModelHookCallCount++;
+            return 'model';
+          }
+        })
+      );
+      this.add(
+        'route:parent.child',
+        Route.extend({
+          queryParams: {
+            childParam: {
+              refreshModel: true
+            }
+          }
+        })
+      );
+      this.add(
+        'controller:parent',
+        Controller.extend({
+          parentParam: 'parentParamDefaultValue'
+        })
+      );
+      this.add(
+        'controller:parent.child',
+        Controller.extend({
+          childParam: 'childParamDefaultValue'
+        })
+      );
+
+      await this.visit('/?parentParam=parentParamDefaultValue');
+      assert.equal(parentModelHookCallCount, 1);
```

Comment:
This looks weird to me. If I'm not missing something, the app should stay in the index route at this time, so I don't think the model hook of the `parent` route should have been called.